### PR TITLE
Add calendar input

### PR DESCRIPTION
**Pull request by Deivis**
Add the an input with calendar selection to control the  one or two dates selection, this component is a simplified version of the `DateInput` component, it removes the presets options and add a custom date blocking function.

#### Interface
```jsx
 <CalendarInput
      dateSelection="single"
      icon={<IconCalendar width={16} height={16} />}
      isValidDay={() => true}
      onChange={() => false}
      months={1}
      strings={{
        end: 'End',
        select: 'Select a date',
        start: 'Initial',
      }}
      value={{
         end: null,
         start: null,
      }}
 />
```

### CODE REVIEW

File: src/Calendar/index.js [hunk 1]
```diff
@@ -34,6 +44,44 @@ const getExtraProps = omit([
...
+    isVisibleEnd = contains(end.month(), months)
+  }
+
+  return {
+    isVisibleEnd,
+    isVisibleStart,
+  }
+}
+
+const needCalendarUpdate = (
```
Comment (LeoNero) [inline src/Calendar/index.js]: vale a pena um ramdinha aqui ou não?

File: src/Calendar/index.js [hunk 2]
```diff
@@ -150,6 +253,12 @@ Calendar.propTypes = {
    * @param {object} dates
    */
   onChange: PropTypes.func.isRequired,
+  /**
+   * This function is trigged when the calendar focus is changed between the
```
Comment (LeoNero) [inline src/Calendar/index.js]: triggered

Comment (LeoNero) [reply src/Calendar/index.js]: I did not understand the `is changed between the visible calendars`

File: src/Calendar/index.js (hunk 1 above)
Comment (Deivis) [reply src/Calendar/index.js]: Suggestions ? 
For me looks like an over engineering uses ramda here.

File: src/Calendar/index.js [hunk 3]
```diff
@@ -34,6 +44,44 @@ const getExtraProps = omit([
...
+    isVisibleStart,
+    isVisibleEnd,
+  }
+) => {
+  if (focusedInput === START_DATE && !isVisibleStart) {
+    return true
+  } else if (!isVisibleEnd) {
+    return true
+  }
+  return false
```
Comment (uselessdev) [inline src/Calendar/index.js]: could be?
```
=> (!isVisibleEnd || (focusedInput === START_DATE && !isVisibleStart))
```

**Comment by uselessdev** [comment]
I can select the same date manually in period selection

screenshot from 2018-07-05 15-38-19

**Pull request merged**